pkg/controller/operators/catalog/operator.go

```diff
@@ -622,10 +622,17 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 	// Registry pod hasn't been created or hasn't been updated since the last configmap update, recreate it
 	logger.Debug("ensuring registry server")
 
-	if err := srcReconciler.EnsureRegistryServer(out); err != nil {
-		syncError = fmt.Errorf("couldn't ensure registry server - %v", err)
-		out.SetError(v1alpha1.CatalogSourceRegistryServerError, syncError)
-		return
+	err = srcReconciler.EnsureRegistryServer(out)
+	if err != nil {
+		if _, ok := err.(reconciler.UpdateNotReadyErr); ok {
+			logger.Debug("requeueing registry server for catalog update check: update pod not yet ready")
+			o.catsrcQueueSet.RequeueAfter(out.GetNamespace(), out.GetName(), 10*time.Second)
```

Comment:
Is there a reason for this 10 seconds requeue in term of duration? Would 10 seconds be too short? It does take a bit of time to pull the index image as it can be pretty big with all the binaries inside. Not sure if we are concerned about requeueing too much? If that is not a concern, than I guess we can ignore this.

Reply:
Also, maybe make this duration a constant so we can adjust it if needed instead of hardcoding.